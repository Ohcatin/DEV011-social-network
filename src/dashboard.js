/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
/* eslint-disable import/named */
/* eslint-disable no-alert */
/* eslint-disable no-console */

import {
  addPost, logout, paintRealTime, giveLike, auth, unGiveLike, deletePost, editPost,
} from './lib';

import edit from './imagen/edit.png';
import eliminar from './imagen/eliminar.png';
import like from './imagen/like.png';
import logotipo from './imagen/logotipo.png';

export function dashboard() {
  // ----- contenedor del nombre red social -----
  const containerDashbord = document.createElement('section');
  containerDashbord.setAttribute('class', 'container-dashbord');
  const nameSocial = document.createElement('header');
  nameSocial.setAttribute('class', 'name-social');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo-dash');
  const welcomeUser = document.createElement('p');
  welcomeUser.setAttribute('class', 'welcome-user');
  const user = auth.currentUser;
  const userEmail = user ? user.displayName || user.email : '';
  const welcomeMessage = document.createElement('p');
  welcomeMessage.setAttribute('class', 'welcome-message');
  welcomeMessage.textContent = `Hola, ${userEmail} <3!`;
  // ----- contenedor publicaciones----
  const postSection = document.createElement('div');
  postSection.setAttribute('class', 'section-post');
  postSection.setAttribute('id', 'post-section');
  //  ----- Contenedor de menu------
  const navigationBar = document.createElement('nav');
  navigationBar.setAttribute('class', 'navigation-bar');
  const listNavigation = document.createElement('ul');
  listNavigation.setAttribute('class', 'list-navigation');
  const liHome = document.createElement('li');
  liHome.setAttribute('class', 'li-home');
  liHome.textContent = 'Inicio';
  const liUpload = document.createElement('li');
  liUpload.setAttribute('class', 'li-upload');
  liUpload.textContent = 'Subir';
  // -----Para subir publicaciones-----
  const form = document.createElement('form');
  form.setAttribute('id', 'postForm');
  form.setAttribute('class', 'content-modal');
  const upload = document.createElement('div');
  upload.setAttribute('class', 'upload');
  upload.textContent = 'Crear nueva publicación';
  const sendComment = document.createElement('input');
  sendComment.setAttribute('class', 'text-input');
  sendComment.setAttribute('id', 'sendComment');
  const buttonSend = document.createElement('button');
  buttonSend.setAttribute('class', 'save-modal');
  buttonSend.setAttribute('id', 'buttonSend');
  buttonSend.setAttribute('type', 'submit');
  const buttonClose = document.createElement('button');
  buttonClose.setAttribute('class', 'close-modal');
  buttonClose.textContent = 'x';
  const liProfile = document.createElement('li');
  liProfile.setAttribute('class', 'li-profile');
  liProfile.textContent = 'Perfil';
  // Botón para cerrar sesión
  const logoutButton = document.createElement('button');
  logoutButton.setAttribute('class', 'logout-button');
  logoutButton.textContent = 'Cerrar Sesión';
  // Crear un div para el modal
  const modal = document.createElement('div');
  modal.classList.add('modal');
  form.appendChild(upload);
  form.appendChild(sendComment);
  form.appendChild(buttonSend);
  form.appendChild(buttonClose);
  // Agregar el formulario al modal
  modal.appendChild(form);
  // para cerrar el modal de subir publicación
  buttonClose.addEventListener('click', () => {
    modal.style.display = 'none';
  });
  // Escuchar evento de envío del formulario
  form.addEventListener('submit', (event) => {
    event.preventDefault();
    const inputPost = form.querySelector('#sendComment');
    const post = inputPost.value;
    // Verificar si el campo de texto no está vacío antes de agregar la publicación
    if (post !== '') {
      // Llamar a la función para agregar la publicación y subir a Firebase
      addPost(post, auth.currentUser.email);
      inputPost.value = '';
    }
    // Cerrar el modal después de enviar la publicación
    modal.style.display = 'none';
  });
  // para modal de form
  liUpload.addEventListener('click', () => {
    modal.style.display = 'block';
    document.body.appendChild(modal);
  });
  // Función para el botón
  logoutButton.addEventListener('click', () => {
    logout();
  });

  img.src = logotipo;
  /* imgTransport.src = 'imagen/transporteBl.png';
  imgHostal.src = 'imagen/alojamientoBl.png';
  imgFood.src = 'imagen/food.png'; */
  buttonSend.textContent = 'Publicar';

  // -----Crear Publicación-----
  paintRealTime((querySnapshot) => {
    postSection.textContent = ' ';
    querySnapshot.forEach((doc) => {
      const postContainer = document.createElement('div');
      postContainer.setAttribute('class', 'post-container');
      const postUser = document.createElement('p');
      postUser.setAttribute('class', 'post-user');
      postUser.textContent = `${doc.data().username || ''}`;
      console.log('name:', doc.data().username || '');
      const postNew = document.createElement('div');
      postNew.setAttribute('class', 'post');
      postNew.setAttribute('id', 'post');
      const reaccion = document.createElement('section');
      reaccion.setAttribute('class', 'reaccion');
      const buttonLike = document.createElement('button');
      buttonLike.setAttribute('class', 'button-border');
      const imgLike = document.createElement('img');
      imgLike.setAttribute('class', 'img-like');
      const buttonEdit = document.createElement('button');
      buttonEdit.setAttribute('class', 'button-border');
      buttonEdit.setAttribute('id', 'edit');
      const imgEdit = document.createElement('img');
      imgEdit.setAttribute('class', 'img-edit');
      const buttonDelete = document.createElement('button');
      buttonDelete.setAttribute('class', 'button-border');
      buttonDelete.setAttribute('id', 'delete');
      const imgDelete = document.createElement('img');
      imgDelete.setAttribute('class', 'img-like');
      const counter = document.createElement('p');
      counter.setAttribute('class', 'counter-like');
      counter.textContent = doc.data().counterLikes.length;
      /* const buttonComment = document.createElement('button');
      buttonComment.setAttribute('class', 'comment');
      const imgComment = document.createElement('img');
      imgComment.setAttribute('class', 'img-comment'); */
      imgLike.src = like;
      imgDelete.src = eliminar;
      imgEdit.src = edit;
      // codigo para que los iconos salgan solo si es tu post
      if (doc.data().email === auth.currentUser.email) {
        imgDelete.style.display = 'block';
      } else {
        imgDelete.style.display = 'none';
      }
      if (doc.data().email === auth.currentUser.email) {
        imgEdit.style.display = 'block';
      } else {
        imgEdit.style.display = 'none';
      }
      postNew.textContent = doc.data().post;

      // ----- Llamar función Like-----
      buttonLike.addEventListener('click', (event) => {
        event.preventDefault();
        const postId = doc.id;
        if (!doc.data().counterLikes.includes(auth.currentUser.email)) {
          giveLike(postId, auth.currentUser.email);
        } else {
          unGiveLike(postId, auth.currentUser.email);
        }
        console.log('data del documento: ', doc.data());
        // counter.textContent = doc.data().counterLikes.length;
        console.log(doc.data().counterLikes.length);
      });
      // ----- Borrar publicación -----
      buttonDelete.addEventListener('click', () => {
        const postId = doc.id;
        if (doc.data().email === auth.currentUser.email) {
          // Pregunta al usuario si realmente quiere eliminar la publicación
          const confirmDelete = window.confirm('¿Estás seguro de que quieres eliminar esta publicación?');

          if (confirmDelete) {
            // Si el usuario confirma, elimina la publicación
            deletePost(postId, auth.currentUser.email);
          } else {
            // Si el usuario cancela, no hagas nada o puedes mostrar un mensaje de cancelación
            console.log('Eliminación cancelada por el usuario');
          }
        } else {
          alert('No puedes eliminar esta publicación');
        }
      });
      // ----------MODAL EDITAR COMENTARIO----------
      const closeModalButton = document.createElement('button');
      closeModalButton.textContent = 'x';
      closeModalButton.setAttribute('class', 'close-modal');
      const editModal = document.createElement('div');
      editModal.classList.add('modal');
      const uploadTextEdit = document.createElement('div');
      uploadTextEdit.setAttribute('class', 'upload');
      uploadTextEdit.textContent = 'Editar publicación';
      const contentModalEdit = document.createElement('div');
      contentModalEdit.setAttribute('class', 'content-modal');
      const editPostContent = document.createElement('input');
      editPostContent.setAttribute('type', 'text');
      editPostContent.setAttribute('class', 'text-input');
      const saveEditButton = document.createElement('button');
      saveEditButton.textContent = 'Guardar';
      saveEditButton.setAttribute('class', 'save-modal');
      // Abre el modal cuando se hace clic en el botón "Editar Publicación"
      buttonEdit.addEventListener('click', () => {
        const currentContent = doc.data().post;
        editModal.style.display = 'block';
        document.body.appendChild(editModal);
        editPostContent.value = currentContent;// Puedes establecer el contenido actual aquí
        console.log(editPostContent);
      });
      // Cierra el modal cuando se hace clic en la "X" o en el fondo oscuro
      closeModalButton.addEventListener('click', () => {
        editModal.style.display = 'none';
      });

      editModal.addEventListener('click', (event) => {
        if (event.target === editModal) {
          editModal.style.display = 'none';
        }
      });

      // Guarda los cambios y cierra el modal
      saveEditButton.addEventListener('click', () => {
        const newContent = editPostContent.value;
        const postId = doc.id;
        if (doc.data().email === auth.currentUser.email) {
          editPost(postId, newContent);
        } else {
          alert('No puedes editar esta publicación');
        }
        editModal.style.display = 'none';
      });
      reaccion.append(buttonLike, counter, buttonEdit, buttonDelete);
      postContainer.append(postUser, postNew, reaccion);
      postSection.append(postContainer);
      buttonLike.append(imgLike);
      buttonDelete.append(imgDelete);
      buttonEdit.append(imgEdit);
      contentModalEdit.append(uploadTextEdit, editPostContent, saveEditButton, closeModalButton);
      editModal.append(contentModalEdit);
    });
  });
  /* filterContainer.append(
    ulFilter,
    filterTransport,
    filterHostal,
    filterRestaurant,
  );
  filterTransport.append(imgTransport);
  filterHostal.append(imgHostal);
  filterRestaurant.append(imgFood);
  nameSocial.append(filterContainer); */
  nameSocial.append(img, welcomeMessage);
  navigationBar.append(listNavigation);
  listNavigation.append(
    liHome,
    liUpload,
    liProfile,
    logoutButton,
  );

  containerDashbord.append(
    nameSocial,
    postSection,
    navigationBar,
  );

  return containerDashbord;
}
