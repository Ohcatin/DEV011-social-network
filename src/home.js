/* eslint-disable no-unused-vars */
/* eslint-disable no-console */
// file home.js
import { auth, signGoogle } from './lib/index.js';
import logo from './imagen/logo-gr.png';

const user = auth.currentUser;
console.log(user);

export function home(navigateTo) {
  // ----- contenedor principal -----
  const principalContainer = document.createElement('section');
  principalContainer.setAttribute('class', 'principal-container');
  // ----- contenedor del nombre red social -----
  const nameSocialContainer = document.createElement('section');
  nameSocialContainer.setAttribute('class', 'name-social-container');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo');
  // ----- contenedor sección home -----
  const sectionHome = document.createElement('section');
  sectionHome.setAttribute('class', 'containerInfo');
  const welcome = document.createElement('h2');
  welcome.setAttribute('class', 'welcome');
  const descriptionSocial = document.createElement('p');
  descriptionSocial.setAttribute('class', 'descriptionSocial');
  const buttonSignIn = document.createElement('button');
  buttonSignIn.setAttribute('class', 'signIn');
  const buttonSignUp = document.createElement('button');
  buttonSignUp.setAttribute('class', 'signUp');
  const connectWith = document.createElement('h6');
  connectWith.setAttribute('class', 'connect-with');
  const openGoogle = document.createElement('button');
  openGoogle.setAttribute('class', 'openGoogle');
  const iconGoogle = document.createElement('span');
  iconGoogle.setAttribute('class', 'icon-google');
  const textGoogle = document.createElement('span');
  textGoogle.setAttribute('class', 'text-google');

  img.src = logo;
  welcome.textContent = 'Bienvenido';
  descriptionSocial.textContent = '¡Tu red social de viajes favorita!';
  connectWith.textContent = 'O conectate con';
  textGoogle.textContent = 'Google';
  buttonSignIn.textContent = 'Iniciar sesión';
  buttonSignIn.addEventListener('click', () => {
    navigateTo('/login');
  });
  buttonSignUp.textContent = 'Registrate';
  buttonSignUp.addEventListener('click', () => {
    navigateTo('/register');
  });
  // Agrega un evento de click para el botón de inicio de sesión con Google
  openGoogle.addEventListener('click', () => {
    signGoogle().then((rest) => navigateTo('/dashboard'));
  });
  openGoogle.append(iconGoogle, textGoogle);
  nameSocialContainer.append(img);
  sectionHome.append(
    welcome,
    descriptionSocial,
    buttonSignIn,
    buttonSignUp,
    connectWith,
    openGoogle,
  );
  principalContainer.append(nameSocialContainer, sectionHome);

  return principalContainer;
}
