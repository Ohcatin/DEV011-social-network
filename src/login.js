/* eslint-disable quotes */
/* eslint-disable no-alert */
/* eslint-disable no-console */
/* eslint-disable no-unused-vars */
// file login finished
import { auth, signGoogle, signIn } from './lib/index.js';
import logoReturn from './imagen/return.png';
import logo from './imagen/logo-gr.png';

function login(navigateTo) {
  // ----- contenedor principal ----
  const principalContainer = document.createElement('section');
  principalContainer.setAttribute('class', 'principal-container');
  const buttonReturn = document.createElement('button');
  buttonReturn.setAttribute('class', 'return');
  const imgReturn = document.createElement('img');
  imgReturn.setAttribute('class', 'return');
  // ----- contenedor nombre red social ----
  const containerName = document.createElement('section');
  containerName.setAttribute('class', 'containerName');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo');
  // ----- contenedor login -----
  const containerSingIn = document.createElement('section');
  containerSingIn.setAttribute('class', 'containerInfo');
  const form = document.createElement('form');
  form.setAttribute('class', 'form-sign');
  const inputEmail = document.createElement('input');
  inputEmail.setAttribute('class', 'inputInfo');
  inputEmail.setAttribute('id', 'idInputEmail');
  inputEmail.setAttribute('autocomplete', 'current-password');
  const inputPass = document.createElement('input');
  inputPass.setAttribute('class', 'inputInfo');
  inputPass.setAttribute('id', 'idInputPass');
  inputPass.setAttribute('type', 'password');
  inputPass.setAttribute('autocomplete', 'current-password');
  const buttonSignIn = document.createElement('button');
  buttonSignIn.setAttribute('class', 'signIn');
  buttonSignIn.setAttribute('id', 'signIn-button');
  // const forgetPassword = document.createElement('input');
  const connectWith = document.createElement('h4');
  connectWith.setAttribute('class', 'connect-with');
  const openGoogle = document.createElement('button');
  openGoogle.setAttribute('class', 'openGoogle');
  const iconGoogle = document.createElement('span');
  iconGoogle.setAttribute('class', 'icon-google');
  const textGoogle = document.createElement('span');
  textGoogle.setAttribute('class', 'text-google');

  img.src = logo;
  imgReturn.src = logoReturn;
  inputEmail.placeholder = 'Correo electrónico';
  inputPass.placeholder = 'Contraseña';
  buttonSignIn.textContent = 'Iniciar sesión';
  connectWith.textContent = 'O conéctate con';
  textGoogle.textContent = 'Google';

  buttonReturn.addEventListener('click', () => {
    navigateTo('/');
  });

  // Agrega un evento de click para el botón de inicio de sesión
  buttonSignIn.addEventListener('click', async () => {
    const email = document.getElementById('idInputEmail').value;
    const password = document.getElementById('idInputPass').value;

    if (!email || !password) {
      // Manejar el caso de campos vacíos, por ejemplo, mostrar un mensaje al usuario
      console.error('Por favor, completa todos los campos.');
      alert('Por favor, completa todos los campos.');
      // Aquí puedes mostrar un mensaje al usuario indicando que ambos campos son obligatorios.
      return; // Detiene la ejecución si hay campos vacíos
    }
    try {
      await signIn(email, password);
      navigateTo('/dashboard');
    } catch (error) {
      // Manejar el error, por ejemplo, mostrar un mensaje al usuario
      // eslint-disable-next-line no-console
      console.error('Error durante el inicio de sesión:', error);
      // Aquí puedes mostrar un mensaje al usuario indicando que ha habido
      // un problema con el inicio de sesión.
    }
  });

  // Agrega un evento de click para el botón de inicio de sesión con Google
  openGoogle.addEventListener('click', () => {
    signGoogle().then((rest) => navigateTo('/dashboard'));
  });

  const user = auth.currentUser;
  if (user !== null) {
    user.providerData.forEach((profile) => {
      // console.log(`Sign-in provider: ${profile.providerId}`);
      // console.log(`  Provider-specific UID: ${profile.uid}`);
      // console.log(`  Name: ${profile.displayName}`);
      // console.log(`  Email: ${profile.email}`);
      // console.log(`  Photo URL: ${profile.photoURL}`);
    });
  }
  openGoogle.append(iconGoogle, textGoogle);
  form.append(inputEmail, inputPass);
  containerName.append(img);
  buttonReturn.append(imgReturn);
  containerSingIn.append(form, buttonSignIn, connectWith, openGoogle);
  principalContainer.append(buttonReturn, containerName, containerSingIn);

  return principalContainer;
}

export default login;
