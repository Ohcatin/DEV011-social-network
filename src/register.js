/* eslint-disable no-alert */
import { registerUser, signGoogle } from './lib';
import logoReturn from './imagen/return.png';
import logo from './imagen/logo-gr.png';

export function register(navigateTo) {
  const principalContainer = document.createElement('section');
  principalContainer.setAttribute('class', 'principal-container');
  const buttonReturn = document.createElement('button');
  buttonReturn.setAttribute('class', 'return');
  buttonReturn.setAttribute('id', 'return');
  const imgReturn = document.createElement('img');
  imgReturn.setAttribute('class', 'return');
  const nameSocialContainer = document.createElement('section');
  nameSocialContainer.setAttribute('class', 'name-social-container');
  const img = document.createElement('img');
  img.setAttribute('class', 'logo');
  // -----Sesión de datos Registro-----
  const registerContainer = document.createElement('section');
  registerContainer.setAttribute('class', 'containerInfo');
  const form = document.createElement('form');
  form.setAttribute('class', 'form-sign');
  const inputName = document.createElement('input');
  inputName.setAttribute('class', 'inputInfo');
  inputName.setAttribute('type', 'text');
  inputName.setAttribute('id', 'idInputName');
  const inputEmail = document.createElement('input');
  inputEmail.setAttribute('class', 'inputInfo');
  inputEmail.setAttribute('type', 'email');
  inputEmail.setAttribute('id', 'idInputEmail');
  const inputPass = document.createElement('input');
  inputPass.setAttribute('class', 'inputInfo');
  inputPass.setAttribute('type', 'password');
  inputPass.setAttribute('id', 'idInputPass');
  const buttonRegister = document.createElement('button');
  buttonRegister.setAttribute('class', 'buttonInfo');
  // -----Sesión Ingresar con Google-----
  const connectWith = document.createElement('h4');
  connectWith.setAttribute('class', 'connect-with');
  const openGoogle = document.createElement('button');
  openGoogle.setAttribute('class', 'openGoogle');
  const iconGoogle = document.createElement('span');
  iconGoogle.setAttribute('class', 'icon-google');
  const textGoogle = document.createElement('span');
  textGoogle.setAttribute('class', 'text-google');

  img.src = logo;
  imgReturn.src = logoReturn;
  inputName.placeholder = 'Nombre Usuario Enrutado';
  inputEmail.placeholder = 'Correo electrónico';
  inputPass.placeholder = 'Contraseña';
  buttonRegister.textContent = 'Registrate';
  connectWith.textContent = 'O conéctate con';
  textGoogle.textContent = 'Google';
  buttonReturn.addEventListener('click', () => {
    navigateTo('/');
  });

  // Evento Sing Up
  buttonRegister.addEventListener('click', async () => {
    const email = document.getElementById('idInputEmail').value;
    const password = document.getElementById('idInputPass').value;
    const name = document.getElementById('idInputName').value;
    registerUser(email, password, name);

    if (!email || !password || !name) {
      alert('Por favor, completa todos los campos.');
      return; // Detiene la ejecución si hay campos vacíos
    }
    try {
      registerUser(email, password, name);
      navigateTo('/dashboard');
    } catch (error) {
      alert('Error durante el inicio de sesión');
    }
  });
  // Agrega un evento de click para el botón de inicio de sesión con Google
  openGoogle.addEventListener('click', () => {
    signGoogle().then(() => navigateTo('/dashboard'));
  });

  openGoogle.append(iconGoogle, textGoogle);
  form.append(inputName, inputEmail, inputPass);
  nameSocialContainer.append(img);
  buttonReturn.append(imgReturn);
  registerContainer.append(form, buttonRegister, connectWith, openGoogle);
  principalContainer.append(buttonReturn, nameSocialContainer, registerContainer);
  // console.log('este es el registro');
  return principalContainer;
}
