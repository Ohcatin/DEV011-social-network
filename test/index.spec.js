/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */

// importamos la funcion que vamos a testear
import { signIn, registerUser } from '../src/lib/index.js';
import { register } from '../src/register.js';
import login from '../src/login.js';
import { dashboard } from '../src/dashboard.js';
import * as index from '../src/lib/index.js';
import { home } from '../src/home.js';

// -----test funcionalidad  register-----
describe('register', () => {
  it('Should be a function', () => {
    expect(typeof register).toBe('function');
  });
  it('Have a button to return', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    const haveAButton = DOM.querySelector('#return');
    expect(haveAButton).not.toBe(undefined);
  });
  it('Snapshot of register', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    expect(DOM).toMatchSnapshot();
  });
  it('Have a button to register with Google', () => {
    const DOM = document.createElement('section');
    DOM.append(register());
    const haveAbuttonGoogle = DOM.querySelector('.openGoogle');
    expect(haveAbuttonGoogle).not.toBe(undefined);
  });
  it('After click button register call function navigateTo with /dashboard', () => {
    const DOM = document.createElement('section');
    document.body.appendChild(DOM);
    const navigateTo = jest.fn();
    DOM.append(register(navigateTo));
    jest.spyOn(index, 'registerUser').mockImplementation(() => Promise.resolve({}));
    const haveAbuttonRegister = DOM.querySelector('.buttonInfo');
    const email = DOM.querySelector('#idInputEmail');
    const password = DOM.querySelector('#idInputPass');
    const name = DOM.querySelector('#idInputName');
    email.value = 'hola@enrutados.com';
    password.value = '123456';
    name.value = 'Labo';
    haveAbuttonRegister.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/dashboard');
  });
});
// -----Test registro de usuario con Spy.Mock-----
describe('have a button signUp to register the user', () => {
  it('test of click button register', () => {
    const DOM = document.createElement('section');
    document.body.appendChild(DOM);
    DOM.append(register());
    jest.spyOn(index, 'registerUser').mockImplementation(() => Promise.resolve({
      message: 'success',
      email: 'hola@enrutados.com',
    }));

    const buttonRegister = DOM.querySelector('.buttonInfo');
    const email = DOM.querySelector('#idInputEmail');
    email.value = 'hola@enrutados.com';
    const password = DOM.querySelector('#idInputPass');
    password.value = '123456';
    buttonRegister.click();
    expect(index.registerUser).toHaveBeenCalledTimes(4);
    // loginSpy.mockRestore();
  });
});

// -----test ruteo funcion registerUser-----//
describe('registerUser', () => {
  it('Should be a function', () => {
    expect(typeof registerUser).toBe('function');
  });
  /* it('After click button register call function navigateTo with /dashboard', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(registerUser(navigateTo));
    const haveAbuttonRegister = DOM.querySelector('.buttonInfo');
    haveAbuttonRegister.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/dashboard');
  }); */
});
// -----test funcionalidad  login-----
describe('login', () => {
  it('Should be a function', () => {
    expect(typeof login).toBe('function');
  });
  it('Have a button to return', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    const haveAButton = DOM.querySelector('.return');
    expect(haveAButton).not.toBe(undefined);
  });
  it('Snapshot of login', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    expect(DOM).toMatchSnapshot();
  });
  it('After click button return call function navigateTo with /', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(login(navigateTo));
    const haveAbuttonReturn = DOM.querySelector('.return');
    haveAbuttonReturn.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/');
  });
  it('Have a button to login with Google', () => {
    const DOM = document.createElement('section');
    DOM.append(login());
    const haveAbuttonGoogle = DOM.querySelector('.openGoogle');
    expect(haveAbuttonGoogle).not.toBe(undefined);
  });
});
describe('signIn', () => {
  it('Should be a function', () => {
    expect(typeof signIn).toBe('function');
  });
});

// ----- test post -----
describe('dashboard', () => {
  it('Should be a function', () => {
    expect(typeof dashboard).toBe('function');
  });
  it('Have a button to log out', () => {
    const DOM = document.createElement('section');
    document.body.appendChild(DOM);
    DOM.append(dashboard());
    const haveAbutton = DOM.querySelector('.logout-button');
    expect(haveAbutton).not.toBe(undefined);
  });
  it('Have a container to posts', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const containerPosts = DOM.querySelector('#post-section');
    expect(containerPosts).not.toBe(undefined);
  });
  it('Have a form to public post', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const createPosts = DOM.querySelector('#postForm');
    expect(createPosts).not.toBe(undefined);
  });
  it('Post have a button to delete', () => {
    const DOM = document.createElement('section');
    DOM.append(dashboard());
    const buttonDelete = DOM.querySelector('#delete');
    expect(buttonDelete).not.toBe(undefined);
  });
  it('Post have a button to edit', () => {
    const DOM = document.createElement('div');
    document.body.append(DOM);
    DOM.append(dashboard());
    const buttonEdit = DOM.querySelector('#edit');
    expect(buttonEdit).not.toBe(undefined);
  });
});
// ----- Funcionalidad de Home -----
describe('home', () => {
  it('Should be a function', () => {
    expect(typeof home).toBe('function');
  });
  it('After click button signIn call function navigateTo with /login', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(home(navigateTo));
    const haveAbuttonSignIn = DOM.querySelector('.signIn');
    haveAbuttonSignIn.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/login');
  });

  it('After click button signUp call function navigateTo with /register', () => {
    const DOM = document.createElement('section');
    const navigateTo = jest.fn();
    DOM.append(home(navigateTo));
    const haveAbuttonSignUp = DOM.querySelector('.signUp');
    haveAbuttonSignUp.click();
    expect(navigateTo).toHaveBeenLastCalledWith('/register');
  });
  it('Snapshot of home', () => {
    const DOM = document.createElement('section');
    DOM.append(home());
    expect(DOM).toMatchSnapshot();
  });
});
