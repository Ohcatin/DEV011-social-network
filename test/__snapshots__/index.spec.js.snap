// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home Snapshot of home 1`] = `
<section>
  <section
    class="principal-container"
  >
    <section
      class="name-social-container"
    >
      <img
        class="logo"
        src="imagen/logo-gr.png"
      />
    </section>
    <section
      class="containerInfo"
    >
      <h2
        class="welcome"
      >
        Bienvenido
      </h2>
      <p
        class="descriptionSocial"
      >
        ¡Tu red social de viajes favorita!
      </p>
      <button
        class="signIn"
      >
        Iniciar sesión
      </button>
      <button
        class="signUp"
      >
        Registrate
      </button>
      <h6
        class="connect-with"
      >
        O conectate con
      </h6>
      <button
        class="openGoogle"
      >
        <span
          class="icon-google"
        />
        <span
          class="text-google"
        >
          Google
        </span>
      </button>
    </section>
  </section>
</section>
`;

exports[`login Snapshot of login 1`] = `
<section>
  <section
    class="principal-container"
  >
    <button
      class="return"
    >
      <img
        class="return"
        src="imagen/return.png"
      />
    </button>
    <section
      class="containerName"
    >
      <img
        class="logo"
        src="imagen/logo-gr.png"
      />
    </section>
    <section
      class="containerInfo"
    >
      <form>
        <input
          autocomplete="current-password"
          class="inputInfo"
          id="idInputEmail"
          placeholder="Correo electrónico"
        />
        <input
          autocomplete="current-password"
          class="inputInfo"
          id="idInputPass"
          placeholder="Contraseña"
          type="password"
        />
      </form>
      <button
        class="signIn"
        id="signIn-button"
      >
        Iniciar sesión
      </button>
      <h4
        class="connect-with"
      >
        O conéctate con
      </h4>
      <button
        class="openGoogle"
      >
        <span
          class="icon-google"
        />
        <span
          class="text-google"
        >
          Google
        </span>
      </button>
    </section>
  </section>
</section>
`;

exports[`register Snapshot of register 1`] = `
<section>
  <section
    class="principal-container"
  >
    <button
      class="return"
      id="return"
    >
      <img
        class="return"
        src="imagen/return.png"
      />
    </button>
    <section
      class="name-social-container"
    >
      <img
        class="logo"
        src="imagen/logo-gr.png"
      />
    </section>
    <section
      class="containerInfo"
    >
      <form>
        <input
          class="inputInfo"
          id="idInputName"
          placeholder="Nombre Usuario"
          type="text"
        />
        <input
          class="inputInfo"
          id="idInputEmail"
          placeholder="Correo electrónico"
          type="email"
        />
        <input
          class="inputInfo"
          id="idInputPass"
          placeholder="Contraseña"
          type="password"
        />
      </form>
      <button
        class="buttonInfo"
      >
        Registrate
      </button>
      <h4
        class="connect-with"
      >
        O conéctate con
      </h4>
      <button
        class="openGoogle"
      >
        <span
          class="icon-google"
        />
        <span
          class="text-google"
        >
          Google
        </span>
      </button>
    </section>
  </section>
</section>
`;
